sigllm/primitives/transformation.py

```diff
@@ -6,34 +6,43 @@
 import numpy as np
 
 
-def format_as_string(X, sep=',', space=False):
+def format_as_string(X, sep=',', space=False, normal=False):
     """Format X to a list of string.
 
-    Transform a 2-D array of integers to a list of strings,
-    seperated by the indicated seperator and space.
+    Transform an array of integers to string(s), separated by the indicated separator and space.
+    Handles two cases:
+    - If normal=True, treats X as a single time series (window_size, 1)
+    - If normal=False, treats X as multiple windows (num_windows, window_size, 1)
 
     Args:
         sep (str):
             String to separate each element in X. Default to `','`.
         space (bool):
             Whether to add space between each digit in the result. Default to `False`.
+        normal (bool):
+            Whether to treat X as a normal time series. If True, expects (window_size, 1)
+            and returns a single string. If False, expects (num_windows, window_size, 1)
+            and returns a list of strings. Default to `False`.
 
     Returns:
-        ndarray:
-            A list of string representation of each row.
+        ndarray or str:
+            If normal=True, returns a single string representation. If normal=False,
+            returns a list of string representations for each window.
     """
 
     def _as_string(x):
         text = sep.join(list(map(str, x.flatten())))
-
         if space:
             text = ' '.join(text)
-
         return text
 
-    results = list(map(_as_string, X))
-
-    return np.array(results)
+    if normal:
+        # Handle as single time series (window_size, 1)
+        return _as_string(X)
+    else:
+        # Handle as multiple windows (num_windows, window_size, 1)
+        results = list(map(_as_string, X))
+        return np.array(results)
```

Comment:
I'm wondering if we should have it more generalized, regardless of normal or not.
```python3
if X.ndim < 2:
    return _as_string(X)
else:
...
```

What do you think?